src/schemathesis/specs/openapi/stateful.py

```diff
@@ -0,0 +1,128 @@
+    target_strategy = endpoint.as_strategy()
+
+    def create_new_case(response: requests.Response) -> SearchStrategy["Case"]:
+        @st.composite  # type: ignore
+        def combined_case(draw: st.DataObject) -> "Case":
+            case = draw(target_strategy)
+            case.path_parameters["user_id"] = response.json()["id"]
+            return case
+
+        return combined_case()
+
+    return create_new_case
+
+
+def call_and_validate(self: RuleBasedStateMachine, case: "Case") -> requests.Response:
+    response = case.call()
+    case.validate_response(response)
+    return response
+
+
+def create_state_machine(schema: "BaseOpenAPISchema", links_field: str) -> Type[RuleBasedStateMachine]:
+    bundles = {}
+    sources: Dict[str, Sources] = defaultdict(lambda: defaultdict(list))
+    for endpoint in schema.get_all_endpoints():
+        name = endpoint.verbose_name
+        bundles[name] = Bundle(name)
+        for status_code, definition in endpoint.definition.resolved["responses"].items():
+            for link in definition.get(links_field, {}).values():
+                if "operationId" in link:
+                    target_endpoint = schema.get_endpoint_by_operation_id(link["operationId"])  # type: ignore
+                else:
+                    target_endpoint = schema.get_endpoint_by_reference(link["operationRef"])  # type: ignore
+                target_name = target_endpoint.verbose_name
+                if status_code == "default":
+                    response_filter = default_status_code(list(endpoint.definition.resolved["responses"]))
+                else:
+                    response_filter = match_status_code(status_code)
+                strategy_builder = make_links_strategy_builder(target_endpoint)
+                sources[target_name][name].append((response_filter, strategy_builder))
+
+    rules = dict(item for endpoint in schema.get_all_endpoints() for item in make_rule(endpoint, sources, bundles))
+
+    return type("StatefulTest", (RuleBasedStateMachine,), {**bundles, **rules})
+
+
+def make_rule(
+    endpoint: "Endpoint", sources: Dict[str, Sources], bundles: Dict[str, Bundle]
+) -> Generator[Tuple[str, Rule], None, None]:
+    name = endpoint.verbose_name
+    if name in sources:
+        strategy = create_strategy(bundles, sources[name])
+    else:
+        target_strategy = endpoint.as_strategy()
+        strategy = target_strategy
+        yield f"initialize {name}", initialize(target=bundles[name], case=target_strategy)(call_and_validate)
+    yield f"rule {name}", rule(target=bundles[name], case=strategy)(call_and_validate)
+
+
+def create_strategy(bundles: Dict[str, Bundle], sources: Sources) -> SearchStrategy["Case"]:
+    """Create a Hypothesis strategy for an endpoint.
+
+    The cases are generated based on responses that are considered "sources" for this endpoint.
+    """
+    strategies = [
+        bundles[source_name].filter(response_filter).flatmap(strategy_builder)
```

Comment:
It could be a function, that takes all bundles and creates a new strategy - this way it could be more flexible - taking one bundle, multiple bundles to combine two responses, etc.